### DISCUSSION TITLE
Provide a configurable ENV variable to include dev tools even in production builds

### DISCUSSION THREAD

**Discussion by steevsachs** [TanStack/query#2849] [Ideas] [answered]
Currently the inclusion of dev tools is hard coded to require NODE_ENV = production. There are cases, such as production builds on internal environments, where exposing dev tools would be extremely useful for debugging live issues.

Is there a workaround to this?  What are thoughts on supporting a variable such as `REACT_QUERY_SHOW_DEVTOOLS_IN_PRODUCTION` that could override the current behavior?

**Comment by TkDodo** [comment, marked as answer]
have a look here at how to lazy-load the development devtools in production, toggle-able by the console:

https://twitter.com/tannerlinsley/status/1392857647037091849

**Reply by steevsachs** [reply]
This definitely works for us as a workaround, thank you!
